### PR TITLE
images: Add needed packages for Cockpit tests

### PR DESCRIPTION
**Pull request by marusak**
* [x] image-refresh rhel4edge

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh rhel4edge done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel4edge-20230202-154458

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/rhel4edge-20230202-154444.log

**Review by martinpitt — DISMISSED** [review]
Not sure if you already want to land this, but looks pretty green to me.

**Comment by marusak** [comment]
There are still [5 failing tests](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18282-20230202-170833-c8aad358-rhel4edge-bots-4351/log.html#26-2). It is possible that I still might need to do some changes to the image as it still does not show insights status for example.

**Comment by cockpituous** [comment]
image-refresh rhel4edge done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel4edge-20230203-135452

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/rhel4edge-20230203-135454.log

**Comment by cockpituous** [comment]
image-refresh rhel4edge done: https://github.com/cockpit-project/bots/commits/image-refresh-rhel4edge-20230203-140559

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/rhel4edge-20230203-140505.log

**Comment by marusak** [comment]
- [test/verify/check-static-login TestLogin.testBasic](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18282-20230202-170833-c8aad358-rhel4edge-bots-4351/log.html#252-2) fails as ws sees the `/etc/issue` from the container (based on f-36) and not the `/etc/issue` from the rhel4edge machine.
- [test/verify/check-connection TestConnection.testTls] fails with `chgrp: invalid group: 'cockpit-ws'`. Indeed, rhel4edge does not have this group at all. However I don't follow why fedora-coreos has it. I though because it contains cockpit-ws and we then remove it and it is left behind but that does not seem to be the case
-

**Comment by martinpitt** [comment]
@marusak: Fedora CoreOS indeed has
```
/usr/lib/group:cockpit-ws:x:987:
/usr/lib/passwd:cockpit-ws:x:988:987:User for cockpit-ws:/:/usr/sbin/nologin
```
This seems to be in the CoreOS rpm-ostree build definition. I don't know right now where that lives, but e.g. it's also present in the [Fedora Workstation config](https://pagure.io/workstation-ostree-config/blob/main/f/group)

**Comment by marusak** [comment]
All except the last one should work in https://github.com/cockpit-project/cockpit/pull/18282

**Comment by marusak** [comment]
journal shows:
```
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.054: cockpit-ssh admin@host.containers.internal: host argument 'admin@host.containers.internal', host 'host.containers.internal', username 'admin', port '0'
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.056: cockpit-ssh admin@host.containers.internal: using known hosts file (null); host known: 1; connect to unknown hosts: 1
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.088: cockpit-ssh admin@host.containers.internal: connected
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.089: cockpit-ssh admin@host.containers.internal: verified host key
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: Public key file /root/.ssh/id_ed25519.pub doesn't exist or isn't readable
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: Public key file /root/.ssh/id_dsa.pub doesn't exist or isn't readable
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: Public key file /root/.ssh/id_ecdsa.pub doesn't exist or isn't readable
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: Public key file /root/.ssh/id_ecdsa_sk.pub doesn't exist or isn't readable
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: Public key file /root/.ssh/id_ed25519_sk.pub doesn't exist or isn't readable
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: Public key file /root/.ssh/id_rsa.pub doesn't exist or isn't readable
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 cockpit-bastion[1013]: (cockpit-ssh:60): cockpit-ssh-DEBUG: 07:55:23.101: cockpit-ssh admin@host.containers.internal: gssapi auth failed
Feb 07 02:55:23 rhel4edge-127-0-0-2-2201 sshd[5360]: Connection closed by authenticating user admin 10.88.0.2 port 50808 [preauth]
```

but there is a key:
```
[root@rhel4edge-127-0-0-2-2201 ~]# ls -l /root/.ssh/
total 8
-rw-r--r--. 1 root root 394 Feb  3 08:57 authorized_keys
-rw-------. 1 root root 394 Feb  7 02:54 id_rsa.pub
```

@martinpitt Any ideas or help how to debug cockpit-ssh?

File: images/scripts/rhel4edge.bootstrap [hunk 1]
```diff
@@ -12,8 +12,13 @@ cat > edge-request.json <<EOF
...
   "packages": [
     { "name": "cockpit-system" },
+    { "name": "insights-client" },
+    { "name": "ncurses" },
+    { "name": "openssh-server" },
     { "name": "socat" },
-    { "name": "openssh-server" }
+    { "name": "sscg" },
+    { "name": "sssd-common" },
+    { "name": "subscription-manager-cockpit" }
```
Comment (martinpitt) [inline images/scripts/rhel4edge.bootstrap:21]: Why do we need to pre-install that? AFAIK we don't use it in cockpit's tests, and doesn't this make it harder to test sub-man PRs, as you would then have to replace already installed packages in the OSTree?

File: images/scripts/rhel4edge.bootstrap [hunk 2]
```diff
@@ -12,8 +12,13 @@ cat > edge-request.json <<EOF
   "imageType": "rhel-edge-installer",
   "packages": [
     { "name": "cockpit-system" },
+    { "name": "insights-client" },
+    { "name": "ncurses" },
```
Comment (martinpitt) [inline images/scripts/rhel4edge.bootstrap:16]: This is rather non-obvious, and needs a "why?" comment.

**Comment by martinpitt** [comment]
@marusak: For the testkeys issue, see https://github.com/cockpit-project/cockpit/pull/18282#issuecomment-1422165638 . So the image is fine as-is, I think, I'd just like to get my above two review comments sorted out.

File: images/scripts/rhel4edge.bootstrap (hunk 1 above)
Comment (marusak) [reply images/scripts/rhel4edge.bootstrap:21]: It is exactly the same what we do in rhel.setup, see https://github.com/cockpit-project/bots/blob/main/images/scripts/rhel.setup#L182

File: images/scripts/rhel4edge.bootstrap (hunk 2 above)
Comment (marusak) [reply images/scripts/rhel4edge.bootstrap:16]: It provides `/usr/bin/clear` which we call in terminal tests: https://github.com/cockpit-project/cockpit/blob/main/test/verify/check-system-terminal#L72

Comment (martinpitt) [reply images/scripts/rhel4edge.bootstrap:16]: I see; do you mind adding that as a comment?

Also, we don't need ncurses for that, `printf "\033c"` is perfectly sufficient for that. We could also just change that in the test.

Comment (marusak) [reply images/scripts/rhel4edge.bootstrap:18]: Added comment.

**Review by martinpitt — APPROVED** [review]
Cheers! Should be good to land now, right?

**Pull request merged**